pace-util/pace/util/initialization/allocator.py

```diff
@@ -54,29 +55,52 @@ def from_backend(cls, sizer: SubtileGridSizer, backend: str):
         numpy = StorageNumpy(backend)
         return cls(sizer, numpy)
 
-    def empty(self, dims: Sequence[str], units: str, dtype: type = float):
+    def empty(
+        self,
+        dims: Sequence[str],
+        units: str,
+        dtype: type = float,
+    ):
         return self._allocate(self._numpy.empty, dims, units, dtype)
 
-    def zeros(self, dims: Sequence[str], units: str, dtype: type = float):
+    def zeros(
+        self,
+        dims: Sequence[str],
+        units: str,
+        dtype: type = float,
+    ):
         return self._allocate(self._numpy.zeros, dims, units, dtype)
 
-    def ones(self, dims: Sequence[str], units: str, dtype: type = float):
+    def ones(
+        self,
+        dims: Sequence[str],
+        units: str,
+        dtype: type = float,
+    ):
         return self._allocate(self._numpy.ones, dims, units, dtype)
 
     def _allocate(
-        self, allocator: Callable, dims: Sequence[str], units: str, dtype: type = float
+        self,
+        allocator: Callable,
+        dims: Sequence[str],
+        units: str,
+        dtype: type = float,
     ):
         origin = self._sizer.get_origin(dims)
         extent = self._sizer.get_extent(dims)
         shape = self._sizer.get_shape(dims)
+        mask = tuple(
+            [
+                any(dim in coord_dims for dim in dims)
+                for coord_dims in [X_DIMS, Y_DIMS, Z_DIMS]
+            ]
+        )
+        spatial_dims = [i for j in [X_DIMS, Y_DIMS, Z_DIMS] for i in j]
+        extra_dims = [i for i in dims if i not in spatial_dims]
+        if len(extra_dims) > 0 or not dims:
+            mask = None
         try:
-            data = allocator(shape, dtype=dtype, default_origin=origin)
+            data = allocator(shape, dtype=dtype, mask=mask)
```

Comment:
Was the default_origin removed intentionally? 
```suggestion
            data = allocator(shape, dtype=dtype, default_origin=origin, mask=mask)
```

Reply:
Great catch, it should not have been deleted.